.github/workflows/release.yml

```
@@ -0,0 +1,98 @@
+name: 'Release'
+
+# Required permissions to: push, publish to GPR, create a PR
+permissions:
+  contents: 'write'
+  packages: 'write'
+  pull-requests: 'write'
+
+# https://github.com/changesets/action#example-workflow
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+on:
+  push:
+    branches:
+      - 'main'
+      # temp for testing
+      - 'test-changesets'
+
+defaults:
+  run:
+    shell: 'bash'
+
+env:
+  # get more verbose logs from the action
+  ACTIONS_RUNNER_DEBUG: true
+
+jobs:
+  release:
+    name: 'Release'
+    if: ${{ github.repository_owner == 'huggingface' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          # This makes Actions fetch all Git history so that Changesets can generate changelogs with the correct commits
+          fetch-depth: 0
+
+      - name: 'Setup pnpm'
+        uses: pnpm/action-setup@v2.2.4
+        with:
+          version: 7.1.5
+
+      - name: 'Setup Node.js [1]'
+        uses: actions/setup-node@v3
+        with:
+          cache: 'pnpm'
+          node-version: 18
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: 'Install dependencies'
+        run: pnpm install
```

```suggestion
        run: pnpm install --filter .
```

Slight optimization, to only install dependencies of the root package